### PR TITLE
Add splines figure and script

### LINKED ISSUE #6643: Improve sample1d -F docs
In order to understand well [sample1d -F](https://docs.generic-mapping-tools.org/dev/sample1d.html#f) I made the following figure.

I think it could be add to the doc to help the users. What do you think? If so, Are you ok with the figure? Any ideas on how to improve it (besides adding a legend and maybe play a bit with the colors)? 

smooth_deriv

```
gmt begin smooth_deriv png
	gmt plot @topo_crossection.txt -Sc0.1c -Gred -N -JX6.5i/2.5i -Bf -Re
	gmt sample1d @topo_crossection.txt -T0.1 -Fl0.0001 > new.txt
	gmt plot new.txt -Wfaint,red
	gmt sample1d @topo_crossection.txt -T0.1 -Fa0.0001 > new.txt
	gmt plot new.txt -W0.25p,blue
	gmt sample1d @topo_crossection.txt -T0.1 -Fc0.0001 > new.txt
	gmt plot new.txt -W0.25p,green
	gmt sample1d @topo_crossection.txt -T0.1 -Fn0.0001 > new.txt
	gmt plot new.txt -W0.25p,orange
	gmt sample1d @topo_crossection.txt -T0.1 -Fs0.0001 > new.txt
	gmt plot new.txt -W0.25p
gmt end
```

### PR DESCRIPTION
**Pull request by Esteban82**
This PR is related to issue #6643. It add a new figure and script to better explain the splines in the docs.

Here I add the script. Let me know if it is ok. 

In what part of the docs should the figure go? At the end of -F argument?

Fixes #6643

### CODE REVIEW

**Comment by Esteban82** [comment]
Thanks Paul. 

Yes, my idea is to edit the rst file and add the figure. I let you know if I need help.

**Comment by Esteban82** [comment]
I am bit confused on how to proceed. Where should I add the ps file?

**Comment by PaulWessel** [comment]
Follow the instructions for adding a new test or doc script on https://docs.generic-mapping-tools.org/dev/devdocs/contributing.html#contributing-documentation

Basically:
Run the full tests
Copy the new PS file to doc/scripts/images
run dvc diff etc etc as it shows above.  This assumes you have set up dvc which I got the impression you had.

**Comment by meghanrjones** [comment]
[This is the specific section](https://docs.generic-mapping-tools.org/dev/devdocs/contributing.html#updating-reference-plots-for-tests) with instructions for adding the figure to `doc/scripts/images`. Please let me know if you think any part of the instructions needs improvement.

**Comment by Esteban82** [comment]
I think I have a problem with running the test. I do it as indicated in docs.

```
1038/1044 Test #1036: test/time/time_testing_6.sh ..................   Passed    1.32 sec
          Start 1042: test/triangulate/nnn.sh
1039/1044 Test #1040: test/trend1d/trend1.sh .......................   Passed    1.26 sec
          Start 1043: test/triangulate/trivor.sh
1040/1044 Test #1042: test/triangulate/nnn.sh ......................   Passed    1.31 sec
          Start 1044: test/xyz2grd/gridrange.sh
1041/1044 Test #1044: test/xyz2grd/gridrange.sh ....................   Passed    0.21 sec
1042/1044 Test #1041: test/trend2d/trend.sh ........................   Passed    1.61 sec
1043/1044 Test #1043: test/triangulate/trivor.sh ...................   Passed    0.95 sec

## It runs until here where it stops for several minutes and I cancel it (Ctrl+c).

^Cmake[3]: *** [CMakeFiles/check.dir/build.make:57: CMakeFiles/check] Interrupción
make[2]: *** [CMakeFiles/Makefile2:660: CMakeFiles/check.dir/all] Interrupción
make[1]: *** [CMakeFiles/Makefile2:667: CMakeFiles/check.dir/rule] Interrupción
make: *** [Makefile:223: check] Interrupción
```

**Comment by meghanrjones** [comment]
> I think I have a problem with running the test. I do it as indicated in docs.
> 
> ```
> 1038/1044 Test #1036: test/time/time_testing_6.sh ..................   Passed    1.32 sec
>           Start 1042: test/triangulate/nnn.sh
> 1039/1044 Test #1040: test/trend1d/trend1.sh .......................   Passed    1.26 sec
>           Start 1043: test/triangulate/trivor.sh
> 1040/1044 Test #1042: test/triangulate/nnn.sh ......................   Passed    1.31 sec
>           Start 1044: test/xyz2grd/gridrange.sh
> 1041/1044 Test #1044: test/xyz2grd/gridrange.sh ....................   Passed    0.21 sec
> 1042/1044 Test #1041: test/trend2d/trend.sh ........................   Passed    1.61 sec
> 1043/1044 Test #1043: test/triangulate/trivor.sh ...................   Passed    0.95 sec
> 
> ## It runs until here where it stops for several minutes and I cancel it (Ctrl+c).
> 
> ^Cmake[3]: *** [CMakeFiles/check.dir/build.make:57: CMakeFiles/check] Interrupción
> make[2]: *** [CMakeFiles/Makefile2:660: CMakeFiles/check.dir/all] Interrupción
> make[1]: *** [CMakeFiles/Makefile2:667: CMakeFiles/check.dir/rule] Interrupción
> make: *** [Makefile:223: check] Interrupción
> ```

Could you try running the tests without any parallelization to find out which test is hanging? I.e., `cd build; ctest` without any `-j` args.

**Comment by joa-quim** [comment]
I have the same problem since 1 or 2 years.

**Comment by seisman** [comment]
@Esteban82 You don't have run the full tests. You can simply run `ctest -R GMT_splines.sh` to run a single test.

**Comment by joa-quim** [comment]
So this also happens in Linux. I though it was only a problem on Windows. The issue in not in the tests themselves but ... somewhere else. Running them individually works but the whole suite hangs before the last two. 
```
ctest -R gridrange
Test project C:/v/build
    Start 1044: test/xyz2grd/gridrange.sh
1/1 Test #1044: test/xyz2grd/gridrange.sh ........   Passed    1.17 sec

100% tests passed, 0 tests failed out of 1
```

**Comment by PaulWessel** [comment]
Script did not have executable permission set.

**Comment by PaulWessel** [comment]
Hi @Esteban82, are you ready to try to upload the PS to dvc and add the lines to sample1d.rst to include the figure and caption?  The plot looks good to me.

**Comment by PaulWessel** [comment]
OK, so I checked out this branch, built it and ran all the test.  I am getting this failure because the orig PS is not yet in the system:

	157 - doc/scripts/GMT_splines.sh (Failed)

I follow the Contribution guide on [test failures](https://docs.generic-mapping-tools.org/dev/devdocs/contributing.html#reviewing-test-failures) and first check that the plot is OK and then copy it to where DVC expects it:

`cp rbuild/doc/scripts/GMT_splines/GMT_splines.ps doc/scripts/images`

Now I can check with dvc:

```
dvc diff
Added:                                                                                                                                           
    doc/scripts/images/GMT_splines.ps

Modified:
    doc/scripts/images/
```

Loos good so I add it to the database:

```
dvc add doc/scripts/images
100% Adding...|█████████████████████████████████████████████████████████████████████████████████████████████████████████|1/1 [00:00,  4.96file/s]
                                                                                                                                                 
To track the changes with git, run:                                                                                                              

    git add doc/scripts/images.dvc

To enable auto staging, run:

	dvc config core.autostage true
```

After that, git tells me that a new file has been staged:

`--- a/doc/scripts/images.dvc`

So I commit my changes to git.

**Comment by PaulWessel** [comment]
OK, had not added the figure and caption to sample1d.rst yet so doing that now.

**Comment by PaulWessel** [comment]
Good, the tests pass and I built the documentation. Here is a screenshot from the relevant new section:

new

Looks good I think - I will remove the WIP tag.

**Comment by PaulWessel** [comment]
Now it is good and ready to be reviewed by all.

File: doc/scripts/GMT_splines.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+6	2.5
+EOF
+# Splines
+gmt begin GMT_splines
+  gmt basemap -R-0.1/6.1/-0.1/3.1 -JX15c/6c -Baf+l"u" -By+l"u(x)" -BWSne
+  gmt sample1d t.txt -T0.01 -Fc	| gmt plot -W1p		  -l"Cubic spline (-Fc)"
+  gmt sample1d t.txt -T0.01 -Fs1	| gmt plot -W1p,orange	  -l"Smooth cubic spline (-Fs1)"
+  gmt sample1d t.txt -T0.01 -Fl	| gmt plot -W1p,blue	  -l"Linear spline (-Fl)"
+  gmt sample1d t.txt -T0.01 -Fa	| gmt plot -W1p,red	  -l"Akima spline (-Fa)"
+  gmt sample1d t.txt -T0.01 -Fn	| gmt plot -W1p,darkgreen -l"Nearest neighbor (-Fn)"
```
Comment (seisman) [inline doc/scripts/GMT_splines.sh:21]: These lines have mixed whitespaces and tabs. It's better to consistently use whitespaces.

**Comment by seisman** [comment]
@PaulWessel When you merged the master branch into this branch, you probably reverted the changes to the dvc file.

**Comment by seisman** [comment]
> Yes, something did not got exactly as planned. I am not sure why but I got a conflict on the dvc file. 

It's because PR #6673 also updated the dvc file.

**Comment by seisman** [comment]
Did you run `git add doc/scripts/images.dvc` again?

**Comment by PaulWessel** [comment]
Just 10 seconds ago.  Should be OK now.

**Comment by Esteban82** [comment]
Many thanks Paui!! I will look into detail this for the next time.

**Comment by Esteban82** [comment]
I was thinking that the script could go to [gmt-examples](https://github.com/GenericMappingTools/gmt-examples). 
What do you think? 

BTW, for gmt-examples, I wonder if would be possible to create like a list of scripts and figure to add on it.

**Pull request merged**